### PR TITLE
Support for init disk with cache and sync mode

### LINKED ISSUE #106: Implement DiskImageStorageDeviceAttachment with mode
**Is your feature request related to a problem? Please describe.**

Implement

- [ ] [`initWithURL:readOnly:cachingMode:synchronizationMode:error:`](https://developer.apple.com/documentation/virtualization/vzdiskimagestoragedeviceattachment/3857729-initwithurl?language=objc)

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by balajiv113**
## Which issue(s) this PR fixes:

Fixes #106

### CODE REVIEW

**Comment by rfay** [comment]
I confim that the workaround in 
* https://github.com/drud/ddev/issues/4466#issuecomment-1361261185
makes a massive positive performance difference for mounting/importing a database, `ddev import-db`

File: storage.go [hunk 1]
```diff
@@ -41,6 +42,28 @@ type DiskImageStorageDeviceAttachment struct {
...
+
+// DiskImageSynchronizationMode describes the disk image synchronization mode.
+//
+// see: https://developer.apple.com/documentation/virtualization/vzdiskimagesynchronizationmode?language=objc
+type DiskImageSynchronizationMode int
+
+const (
+	DiskImageSynchronizationModeFull  DiskImageSynchronizationMode = 1
+	DiskImageSynchronizationModeFsync DiskImageSynchronizationMode = 2
+	DiskImageSynchronizationModeNone  DiskImageSynchronizationMode = 3
```
Comment (Code-Hex) [inline storage.go:64]: This is a matter of taste. I like

```suggestion
	DiskImageSynchronizationModeFull  DiskImageSynchronizationMode = 1 + iota
	DiskImageSynchronizationModeFsync DiskImageSynchronizationMode
	DiskImageSynchronizationModeNone  DiskImageSynchronizationMode
```

File: storage.go [hunk 2]
```diff
@@ -79,6 +102,48 @@ func NewDiskImageStorageDeviceAttachment(diskPath string, readOnly bool) (*DiskI
...
+// Returns error is not nil, assigned with the error if the initialization failed.
+//
+// - diskPath is local file URL to the disk image in RAW format.
+// - readOnly if YES, the device attachment is read-only, otherwise the device can write data to the disk image.
+// - cachingMode is one of the available DiskImageCachingMode options.
+// - syncMode is to define how the disk image synchronizes with the underlying storage when the guest operating system flushes data, described by one of the available DiskImageSynchronizationMode modes.
+//
+// This is only supported on macOS 12 and newer, error will
+// be returned on older versions.
+func NewDiskImageStorageDeviceAttachmentWithCacheAndSync(diskPath string, readOnly bool, cachingMode DiskImageCachingMode, syncMode DiskImageSynchronizationMode) (*DiskImageStorageDeviceAttachment, error) {
```
Comment (Code-Hex) [inline storage.go:115]: Could you add tests here?

This is to check availability.
https://github.com/Code-Hex/vz/blob/23eeededdc039cdcbab348836d84432891aea7ca/osversion_test.go#L109

This one is for checking actually working or not. (It's OK only check exists disk path)
https://github.com/Code-Hex/vz/blob/main/storage_test.go

**Review by Code-Hex — COMMENTED** [review]
Thanks. almost LGTM

**Comment by Code-Hex** [comment]
@rfay If you have any information on what to enable in the Linux Kernel Config (like `CONFIG_*`) to use this feature, I would be very grateful if you could share it with me.

I'm writing wiki that consolidates all this information.
https://github.com/Code-Hex/vz/wiki

**Comment by rfay** [comment]
I'm afraid I'm pretty ignorant; just tried out @balajiv113 's comment/workaround and was super happy with the result.

File: storage.go (hunk 2 above)
Comment (balajiv113) [reply storage.go:115]: Done the first test, about second test we don't have disk path from configuration i believe.
I can add a test to create disk (with cache and sync mode) and start vm and see if its running. Is that okay ?

**Comment by balajiv113** [comment]
> Linux Kernel Config

@Code-Hex 
On this unfortunately I couldn't find a equivalent kernel config. I thought using DiskImageSynchronizationModeNone should be equivalent to this workaround. But on testing looks like its not. 

DiskImageSynchronizationModeFsync and DiskImageSynchronizationModeNone performing more or less same.

File: storage.go (hunk 1 above)
Comment (balajiv113) [reply storage.go]: Done

File: storage.go (hunk 2 above)
Comment (Code-Hex) [reply storage.go:115]: @balajiv113 Sure! Thanks!

**Comment by Code-Hex** [comment]
@balajiv113 I see. Thank you for your comments!

**Review by Code-Hex — APPROVED** [review]
LGTM

**Comment by balajiv113** [comment]
Updated PR with verified commit as well :)

**Pull request merged**